addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -123,7 +124,7 @@ const Wysiwyg = Widget.extend({
 
         this.$editable = this.$editable || this.$el;
         if (options.value) {
-            this.$editable.html(options.value);
+            this.$editable.html(weEditorUtils.xmlToHtml(options.value));
```

Comment:
I think this uses jQuery for historical reasons and we never bothered to change it even when we did make changes around that line. I think you could forgo the `xmlToHtml` util altogether and bypass jQuery by doing `this.$editable[0].innerHTML = options.value;`. Let me know if it solves your issue or if there are other shenanigans going about. cc @Goaman Since it is very likely I'll be on holiday by the time.

Reply:
I'm fine with getting rid of jquery but I still need xmlToHtml as outlined above

Jquery was what fixed it, the default browser behavior is what breaks stuff.

Reply:
Sorry we did not test `innerHTML` properly. It indeed creates the issue.

Reply:
Sorry for the misunderstanding. However, this line here reintroduces the XSS that was fixed in jQuery 3.5. Basically if `options.value` is "<script>alert(Gotcha !)" then the script will be executed. This can be fixed if you use `innerHTML` rather than jQuery `.html` here, but as an additional protection measure I suggest to use `DOMPurify` to sanitize the return value of `xmlToHtml` in the util function itself so that nobody fall into this trap again.

Reply:
This is actually super dodgy to fix in the front-end now that you show me that... Because dompurify is going to break the 'raw templates' every single time. In the back-end we're sure it's at least well-formed xml so it's fine but here anything goes.

I'll see with xmo I guess...